### DISCUSSION TITLE
[ Feature Request ] Show progress names next to the bar

### DISCUSSION THREAD

**Discussion by OliverDudgeon** [fonsp/Pluto.jl#2772] [Pluto development]
Hi, many thanks for developing Pluto.jl. It's of great use in my research.

The support for ProgressLogging.jl is very useful to me. Often I have several progress bars for different parts of a program (often nested inside a package, so not easily separable into multiple cells). Currently, it seems that the name from the name kwarg to the `@progress` or `@withprogress` macros, e.g.

```julia
@progress name = "iterating..." for i in 1:100
	sleep(0.1)
end
```

isn't displayed inline, which is can make it difficult to figure out what is currently running. I tried to figure this out in the preact code but I got quite confused with the custom elements which I'm not used to in the react code I've written. Presumably, this info is available to the frontend and could be displayed inline next to the progress bar?

**Comment by Pangoraw** [comment]
With preact, components are not written as JSX syntax but interpoled in the html string macro. This is the progress component:

https://github.com/fonsp/Pluto.jl/blob/be51e4bf8ceaf14bbc5977892a9ac74afefb54c0/frontend/components/Logs.js#L91-L100

Which is invoked here depending on the whether or not the log message is a progress log, where progress is extracted from the log's kwargs:

https://github.com/fonsp/Pluto.jl/blob/be51e4bf8ceaf14bbc5977892a9ac74afefb54c0/frontend/components/Logs.js#L134-L135

**Comment by OliverDudgeon** [comment]
This seems to be implemented by #2966
